### PR TITLE
Make protected_packages an append options

### PR DESCRIPTION
**Pull request by j-mracek**
It will allow to use configuration overrides from /usr for protected packages.

Resolves: https://github.com/rpm-software-management/dnf5/issues/1107

CI: https://github.com/rpm-software-management/ci-dnf-stack/pull/1431

### CODE REVIEW

**Review by Conan-Kudo — DISMISSED** [review]

File: doc/dnf5.conf.5.rst [hunk 1]
```diff
@@ -373,7 +373,7 @@ repository configuration file should aside from repo ID consists of baseurl, met
 ``protected_packages``
     :ref:`list <list-label>`
 
-    List of packages that DNF5 should never completely remove.
+    This append list option contains name of packages that DNF5 should never completely remove.
```
Comment (keszybz) [inline doc/dnf5.conf.5.rst:376]: "names"

**Review by keszybz — COMMENTED** [review]
I can't comment on the details of the C++ code, but at the high level, this looks like it should do the right thing.

**Review by Conan-Kudo — APPROVED** [review]

**Comment by Conan-Kudo** [comment]
@j-mracek should I merge the CI PR first and re-run this? Or merge this and then the CI PR?

**Comment by j-mracek** [comment]
@Conan-Kudo I have to fix a typo first.

Our CI is broken, therefore we have to run all tests locally. Download ci-dnf-stack project build or download packages with patch to `rpms` directory. Then run `./container-test build --base fedora:38 && ./container-test run --command=dnf5 --tag dnf5`, where `fedora:38` should be modify according to rpms with the patch.

**Pull request merged**